komodoenv/__main__.py

```diff
@@ -0,0 +1,153 @@
+    name = os.environ.get("KOMODO_RELEASE", "")
+    if not name:
+        return None
+
+    name = os.environ["KOMODO_RELEASE"]
+
+    release_root = root / os.environ["KOMODO_RELEASE"]
+    if not release_root.is_dir():
+        return None
+
+    for mode in "stable", "testing", "unstable", "bleeding":
+        for suffix in "", "-py2", "-py27", "-py3", "-py36":
+            dir_ = root / (mode + suffix)
+            if not dir_.is_dir():
+                continue
+            symlink = dir_.resolve()
+            if symlink.name == name:
+                return symlink
+
+    return None
+
+
+def parse_args(args):
+    ap = argparse.ArgumentParser()
+    ap.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        default=False,
+        help="Overwrite exising komodoenv",
+    )
+    ap.add_argument(
+        "-r",
+        "--release",
+        type=str,
+        required=False,
+        help="Komodo release to base komodoenv on. Defaults to currently active komodo release",
+    )
+    ap.add_argument(
+        "--root",
+        type=str,
+        default="/prog/res/komodo",
+        help="Absolute path to komodo root (default: /prog/res/komodo)",
+    )
+    ap.add_argument(
+        "--force-color", action="store_true", default=False, help="Force color output"
+    )
+    ap.add_argument("destination", type=str, help="Where to create komodoenv")
+
+    args = ap.parse_args(args)
+
+    args.root = Path(args.root)
+    assert args.root.is_dir()
+
+    if args.release is None:
+        args.release = autodetect(args.root)
+    elif "/" in args.release:
+        args.release = Path(args.release)
+    else:
+        args.release = args.root / args.release
+
+    args.destination = Path(args.destination).absolute()
+
+    if args.release is None or not args.release.is_dir():
+        logging.error(
+            "Could not automatically detect active Komodo release. "
+            "Either enable a Komodo release that supports komodoenv "
+            "or specify release manually with the "
+            "`--release' argument. "
+        )
+        ap.print_help()
+        sys.exit(1)
+
+    return args
+
+
+def main(args=None):
+    texts = {
+        "beta": red(
+            "Komodoenv is still in beta. Be aware that issues might occur and "
+            "recreating environments once in a while is necessary.\n"
+        ),
+        "nfs": yellow(
+            "Warning: Komodoenv target directory is not located on an NFS "
+            "filesystem. Be aware that multi-machine workloads via eg. LFS "
+            "might not work correctly.\n"
+        ),
+    }
+
+    if args is None:
+        args = sys.argv[1:]
+    args = parse_args(args)
+
+    if args.destination.is_dir() and args.force:
+        rmtree(str(args.destination), ignore_errors=True)
+    elif args.destination.is_dir():
+        print("Destination directory already exists: {}".format(args.destination))
```

Comment:
```suggestion
        sys.exit("Destination directory already exists: {}".format(args.destination))
```

Reply:
Woah. I didn't know this. Mind is blown.